### DISCUSSION TITLE
Dynamic proxy with Nginx Unit

### DISCUSSION THREAD

**Discussion by uhlhosting** [nginx/unit#1358] [Support] [answered]
I want to run something like 

`curl https://my-nginx-unit.com/?url=https://my-proxied-url.com`

and return the content of https://my-proxied-url.com (url will always be different)
the point is to have clients whitelist the IP of http://my-nginx-unit.com in their servers
it should basically emulate what a lightweigth squid server would do

**Comment by lcrilly** [comment, marked as answer]
Unit cannot solve this because

1. The HTTPS scheme is not supported.
2. `proxy` targets must be an IP address (or Unix socket) as hostname DNS lookup is not supported.

**Reply by skauk** [reply]
Are there plans to support DNS lookup for proxying?

**Reply by lcrilly** [reply]
Not at this time. https://github.com/nginx/unit/issues/779#issuecomment-2075467029
